managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/Futures.java

```diff
@@ -68,4 +69,28 @@ public static CompletableFuture<Void> waitForAll(List<CompletableFuture<Void>> f
 
         return compositeFuture;
     }
+
+    public static <T> CompletableFuture<T> executeWithRetry(Supplier<CompletableFuture<T>> op,
+                                                            Class<? extends Exception> needRetryExceptionClass) {
+        CompletableFuture<T> resultFuture = new CompletableFuture<>();
+        op.get().whenComplete((res, ex) -> {
+            if (ex == null) {
+                resultFuture.complete(res);
+            } else {
+                if (needRetryExceptionClass.isAssignableFrom(ex.getClass())) {
+                    executeWithRetry(op, needRetryExceptionClass).whenComplete((res2, ex2) -> {
+                        if (ex2 == null) {
+                            resultFuture.complete(res2);
+                        } else {
+                            resultFuture.completeExceptionally(ex2);
+                        }
+                    });
+                    return;
+                }
+                resultFuture.completeExceptionally(ex);
+            }
+        });
+
+        return resultFuture;
+    }
```

Comment:
Retry without any backoff is not usually recommended. could we use the exponential backoff solution here?

Reply:
Maybe we don't need to introduce a new method. the `RetryUtil#retryAsynchronously` method maybe can work here.

Reply:
`RetryUtil#retryAsynchronously` can't retry based on a specific exception, I will improve it in another PR when.